notifications/smtp/smtp.go

```diff
@@ -0,0 +1,114 @@
+package smtp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"mime/multipart"
+	"net/mail"
+	"net/smtp"
+	"net/textproto"
+
+	"github.com/vocdoni/saas-backend/notifications"
+)
+
+type SMTPConfig struct {
+	FromName     string
+	FromAddress  string
+	SMTPServer   string
+	SMTPPort     int
+	SMTPUsername string
+	SMTPPassword string
+}
+
+type SMTPEmail struct {
+	config *SMTPConfig
+	auth   smtp.Auth
+}
+
+func (se *SMTPEmail) Init(rawConfig any) error {
+	// parse configuration
+	config, ok := rawConfig.(*SMTPConfig)
+	if !ok {
+		return fmt.Errorf("invalid SMTP configuration")
+	}
+	// parse from email
+	if _, err := mail.ParseAddress(config.FromAddress); err != nil {
+		return fmt.Errorf("could not parse from email: %v", err)
+	}
+	// set configuration in struct
+	se.config = config
+	// init SMTP auth
+	se.auth = smtp.PlainAuth("", se.config.SMTPUsername, se.config.SMTPPassword, se.config.SMTPServer)
```

Comment:
The SMTP server could allow no authentication (i.e for local testing using local sendmail). So auth should be only configured if user/password are defined, and they should not be enforced to exist.

Reply:
Updated with new methods that are also required for testing.